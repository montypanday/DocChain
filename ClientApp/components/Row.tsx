import * as React from 'react';
import { render } from 'react-dom';
//import { Preview } from '../components/filePreview';
//var Modal = ReactBootstrap.Modal;
import { Modal } from 'react-bootstrap';


export interface AppProps {
    filename: string,
    size: string,
    lastModified: string,
    downloadUrl: string,
    //isOpen: boolean
}

export interface AppState {

}

export class Row extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super();
        this.showPreviewOrNavigate = this.showPreviewOrNavigate.bind(this);
        this.state = {
            showModal: false
        }
    }

    //toggleModal() {
    //    this.setState({ isOpen: true });
    //}

    getFileExtension(filename) {
        return filename.slice((filename.lastIndexOf(".") - 1 >>> 0) + 2);
    }

    getIconClass(extension) {
        switch (extension) {
            case 'txt':
                return "fa fa-file-text-o fa-2x";
            case 'docx':
                return "fa fa-file-word-o fa-2x";
            case 'zip':
                return "fa fa-file-archive-o fa-2x";
            case 'pptx':
                return "fa fa-file-powerpoint-o fa-2x";
            case 'm4a':
                return "fa fa-file-audio-o fa-2x";
            case 'png':
                return "fa fa-file-image-o fa-2x";
            case 'PNG':
                return "fa fa-file-image-o fa-2x";
            case 'pdf':
                return "fa fa-file-pdf-o fa-2x";
            case 'jpg':
                return "fa fa-file-image-o fa-2x";
            case 'gif':
                return "fa fa-file-image-o fa-2x";
            case 'xlsx':
                return "fa fa-file-excel-o fa-2x";
            case '':
                return "fa fa-folder fa-2x";
            default:
                return "fa fa-file fa-2x";
        }
    }

    showPreviewOrNavigate() {
        console.log("showPreviewOrNavigate is clicked");
        this.setState({ showModal: true });
    }
    render() {
        var a = this.getFileExtension(this.props.filename);
        var iconClass = this.getIconClass(a);
        return (
            <tr>

                <td className="col-xs-6 "><span className={iconClass} style={{ verticalAlign: 'middle', float: 'left' }}></span><h5 onClick={this.showPreviewOrNavigate} style={{ float: 'left', paddingLeft: '15px' }}><div style={{ cursor: 'pointer' }}>{this.props.filename}</div></h5></td>
                <td className="col-xs-1 ">
                    <div className="dropdown">
                        <button className="btn btn-default dropdown-toggle" style={{ verticalAlign: 'middle' }} type="button" data-toggle="dropdown">...
                            </button>
                        <ul className="dropdown-menu">
                            <li><a href={this.props.downloadUrl} download><i className="fa fa-download" aria-hidden="true"></i>       Download</a></li>
                            <li><a href="#" >Preview</a></li>
                            <li><a href="#">JavaScript</a></li>
                            <li className="divider"></li>
                            <li><a href="#">About Us</a></li>
                        </ul>
                    </div>
                </td>
                <td className="col-xs-1 " style={{ verticalAlign: 'middle' }}><i className="fa fa-lock fa-2x"></i></td>
                <td className="col-xs-1 " style={{ verticalAlign: 'middle' }}>{this.props.size}</td>
                <td className="col-xs-2 " style={{ verticalAlign: 'middle' }}>{this.props.lastModified}</td>
            </tr>
        );
    }
}
//<td><i className="fa fa-exclamation-triangle fa-2x"></i></td>