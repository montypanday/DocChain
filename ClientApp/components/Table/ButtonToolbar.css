div {
    font-family: 'europa', sans-serif;
}

.power-icon{
    padding-right: 5px;
}