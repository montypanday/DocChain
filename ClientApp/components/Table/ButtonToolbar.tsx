import * as React from "react";
import { render } from "react-dom";
import { ButtonToolbar } from "react-bootstrap";
import { Button, DropdownButton, MenuItem } from "react-bootstrap";

export interface ButtonToolbarProps {
    NewFolderHandler: any;
    uploadHandler: any;
}

export interface ButtonToolbarState {
}

export default class ButtonToolBar extends React.Component<ButtonToolbarProps, ButtonToolbarState> {
    constructor(props: ButtonToolbarProps) {
        super(props);

        this.state = {
        };
    }
    shouldComponentUpdate(nextProps) {
        return false;
    }

    render() {
        console.log("Button Toolbar was rendered again");
        return (

            <ButtonToolbar>
                <div className="btn-group">
                    <button type="button" className="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                        <i style={{ paddingRight: "4px" }} className="fa fa-upload"></i>
                        Upload <span className="caret"></span>
                    </button>
                    <ul className="dropdown-menu">
                        <li><a className="btn-file">
                            Upload Files <input id="uploadINPUT" multiple type="file" onChange={e => { this.props.uploadHandler(e.target.files); }} />
                        </a></li>
                    </ul>
                </div>
                <div className="btn-group">
                    <button type="button" className="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                        New <span className="caret"></span>
                    </button>
                    <ul className="dropdown-menu">
                        {/* // TODO: PATRICK: Add icon for dropdown options.
                         // For example New Folder option should show a fontawesome folder in left*/}
                        <li><a onClick={this.props.NewFolderHandler} > New Folder</a></li>
                    </ul>
                </div>
                <Button bsStyle="danger"><i className="fa fa-power-off power-icon" aria-hidden="true"></i>Logout</Button>
            </ButtonToolbar>

        );
    }
}