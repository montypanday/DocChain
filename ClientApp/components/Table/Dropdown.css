.dropdown-menu i{
    padding-right: 5px
}

.dropdown-menu li {
    cursor: pointer;
}