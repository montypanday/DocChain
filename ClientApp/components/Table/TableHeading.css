thead {
    font-family: 'europa', sans-serif;
}
