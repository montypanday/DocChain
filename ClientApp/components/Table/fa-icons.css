/* FILE ICON STYLING */
.fa-lock {
    color: rgb(108,187,255);
}

.fa-file-pdf-o {
    color: red;
}

.fa-file-powerpoint-o {
    color: coral;
}

.fa-file-word-o {
    color: dodgerblue;
}

.fa-file-image-o {
    color: slategrey;
}


.fa-file-excel-o {
    color: forestgreen;
}

.fa-folder {
    color: #ffcc80;
}

.fa-file-archive-o {
    color: #ffcc80;
}