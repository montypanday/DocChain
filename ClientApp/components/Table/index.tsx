
import { Row } from "./Row";
import TableHeading from "./TableHeading";

export { Row, TableHeading };