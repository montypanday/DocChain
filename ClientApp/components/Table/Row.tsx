import * as React from 'react';
import { render } from 'react-dom';
require('./Dropdown.css');
require('./fa-icons.css');


export interface AppProps {
    type: string,
    filename: string,
    size: string,
    lastModified: string,
    downloadUrl: string,
    navHandler: any
    id: any
    mimeType: any,
    iconLink: any,
    deleteHandler: any
}

export interface AppState {
}

export class Row extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {
        }
    }

    getFileExtension(filename) {
        try {
            return filename.slice((filename.lastIndexOf(".") - 1 >>> 0) + 2);
        }
        catch{
            return "";
        }
    }

    getIconClass(extension, mimeType) {

        const FormatDictinary = {
            "txt": "text-o",
            "doc": "word-o",
            "docs": "word-o",
            "docx": "word-o",
            "gdoc": "word-o",
            "pdf": "pdf-o",
            "xls": "excel-o",
            "xlsx": "excel-o",
            "gsheets": "excel-o",
            "ppt": "powerpoint-o",
            "pptx": "powerpoint-o",
            "gslides": "powerpoint-o",
            "m4a": "audio-o",
            "mp4": "audio-o",
            "mp3": "audio-o",
            "png": "image-o",
            "PNG": "image-o",
            "JPG": "image-o",
            "jpg": "image-o",
            "JPEG": "image-o",
            "jpeg": "image-o",
            "gif": "image-o",
            "GIF": "image-o",
            "cs": "code-o",
            "zip": "archive-o",
            "flv": "video-o"
        }

        const GoogleFormatDictionary = {
            "application/vnd.google-apps.spreadsheet": "fa fa-2x fa-file-excel-o",
            "application/vnd.google-apps.document": "fa fa-2x fa-file-word-o",
            "application/vnd.openxmlformats-officedocument.wordprocessingml.document": "fa fa-2x fa-file-word-o",
            "application/vnd.google-apps.presentation": "fa fa-2x fa-file-powerpoint-o",
            "application/vnd.google-apps.drawing": "fa fa-2x fa-picture-o",
            "application/vnd.google-apps.map": "fa fa-2x fa-map-marker",
            "application/vnd.google-apps.form": "fa fa-2x fa-wpforms",
            "application/vnd.google-apps.site": "fa fa-2x fa-html5",
            "image/jpeg": "fa fa-2x fa-file-picture-o"
        }

        if (FormatDictinary[extension]) {
            return "fa fa-2x fa-file-" + FormatDictinary[extension];
        }
        if (extension === "" && mimeType === "") {
            return "fa fa-folder fa-2x";
        }
        if (GoogleFormatDictionary[mimeType]) {
            return GoogleFormatDictionary[mimeType];
        }
        if (mimeType == "application/vnd.google-apps.folder") {
            return "fa fa-folder fa-2x";
        }
        return "fa fa-2x fa-file-o";
    }

    doSomething(e) {
        console.log("something");

        var fileActionJson = {
            fileAction: {
                "FileID": "11111111",
                "FileHash": "256bit hexadecimal number",
                "StoragePlatform": "Docchain Development",
                "UserID": "22222222",
                "ActionType": "Test Action"
            }
        }

        fetch('https://localhost:44374/api/FileAction/LogAction', {
            method: 'POST',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(fileActionJson)
        })
    }

    shouldComponentUpdate(nextProps) {
        const a = this.props.type !== nextProps.type;
        const b = this.props.filename !== nextProps.filename;
        const c = this.props.size !== nextProps.size;
        const d = this.props.lastModified !== nextProps.lastModified;
        return (a || b || c || d);
    }

    render() {
        console.log("Row was renderered again");
        var a = this.getFileExtension(this.props.filename);
        var iconClass = this.getIconClass(a, this.props.mimeType);
        var icon;
        icon = <span className={iconClass} style={{ verticalAlign: 'middle', float: 'left', fontSize: '2em' }}></span>;
        return (
            <tr>
                <td className="col-xs-6 " >
                    <a onClick={this.props.navHandler} style={{ cursor: 'pointer' }}>
                        {icon}
                        <h5 style={{ float: 'left', paddingLeft: '15px' }}>{this.props.filename}</h5>
                    </a>
                </td>
                <td className="col-xs-1 ">
                    {this.props.id != "sharedWithMe" && <div className="dropdown">
                        <button className="btn btn-default dropdown-toggle" style={{ verticalAlign: 'middle' }} type="button" data-toggle="dropdown">...
                            </button>
                        <ul className="dropdown-menu">
                            <li><a href={this.props.downloadUrl} download><i className="fa fa-download dropDownIcon" aria-hidden="true"></i>Download</a></li>
                            {this.props.type != 'folder' && <li><a onClick={this.props.navHandler} ><i className="fa fa-eye dropDownIcon" aria-hidden="true"></i>Preview</a></li>}
                            {this.props.id != "sharedWithMe" && < li > <a onClick={this.props.deleteHandler}><i className="fa fa-trash-o dropDownIcon" aria-hidden="true"></i>Delete</a></li>}
                            <li><a href="#"><i className="fa fa-pencil-square-o dropDownIcon" aria-hidden="true"></i>Rename</a></li>
                            <li><a href="#"><i className="fa fa-share dropDownIcon" aria-hidden="true"></i>Share</a></li>

                            {this.props.type != 'folder' && <li className="divider"></li>}
                            {this.props.type != 'folder' && <li><a href="#"><i className="fa fa-database dropDownIcon" aria-hidden="true"></i>Get Document Trail</a></li>}
                            {this.props.type != 'folder' && <li><a href="#"><i className="fa fa-play dropDownIcon" aria-hidden="true"></i>Start Trail</a></li>}
                            {this.props.type != 'folder' && <li><a href="#"><i className="fa fa-stop dropDownIcon" aria-hidden="true"></i>Stop Trail</a></li>}
                            {this.props.type != 'folder' && <li><a href="#"><i className="fa fa-certificate dropDownIcon" aria-hidden="true"></i>Get DocChain Certificate</a></li>}
                            {this.props.type != 'folder' && <li><a href="#"><i className="fa fa-link dropDownIcon" aria-hidden="true"></i>Embed Document</a></li>}
                            {this.props.type != 'folder' && <li><a href="#"><i className="fa fa-calendar-check-o dropDownIcon" aria-hidden="true"></i>Check File</a></li>}
                        </ul>
                    </div>}
                </td>
                {this.props.type != 'folder' && <td className="col-xs-1 " style={{ verticalAlign: 'middle' }}><i className="fa fa-lock fa-2x"></i></td>}
                <td className="col-xs-1 " style={{ verticalAlign: 'middle' }}>{this.props.size}</td>
                <td className="col-xs-2 " style={{ verticalAlign: 'middle' }}>{this.props.lastModified}</td>
            </tr>
        );
    }
}
//<td><i className="fa fa-exclamation-triangle fa-2x"></i></td>