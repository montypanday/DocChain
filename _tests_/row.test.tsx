

import * as React from 'react';
import { Row } from '../ClientApp/components/Table/Row';
import * as renderer from 'react-test-renderer';


test('Link changes the class when hovered', () => {
    const testnavHandler = jest.fn();
    const test_delete_handler = jest.fn();
    const component = renderer.create(
        <Row filename="sample folder"
            id="43970020269"
            lastModified="1/01/2018 6:46:08 PM"
            mimeType=""
            size="28528"
            type="folder"
            key="43970020269"
            navHandler={testnavHandler}

            renameHandler=""
            shareLinkHandler=""

            deleteHandler={test_delete_handler}
            platform={"testPlatform"}

        ></Row>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    //// manually trigger the callback
    //tree.props.onMouseEnter();
    //// re-rendering
    //tree = component.toJSON();
    //expect(tree).toMatchSnapshot();

    //// manually trigger the callback
    //tree.props.onMouseLeave();
    //// re-rendering
    //tree = component.toJSON();
    //expect(tree).toMatchSnapshot();
});