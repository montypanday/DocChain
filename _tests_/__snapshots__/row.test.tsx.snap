// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Whether the componet is right rendered 1`] = `
<tr
  id="43970020269"
>
  <td
    className="col-xs-6 "
    id="43970020269"
  >
    <a
      onClick=""
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <span
        className="fa fa-folder fa-2x"
        style={
          Object {
            "float": "left",
            "fontSize": "1.5em",
            "verticalAlign": "middle",
          }
        }
      />
      <p
        style={
          Object {
            "float": "left",
            "paddingLeft": "15px",
          }
        }
      >
        sample folder
      </p>
    </a>
  </td>
  <td
    className="col-xs-1 "
  >
    <div
      className="dropdown"
    >
      <button
        className="btn btn-default btn-line dropdown-toggle action-btn"
        data-toggle="dropdown"
        style={
          Object {
            "verticalAlign": "middle",
            "zIndex": 0,
          }
        }
        type="button"
      >
        ...
      </button>
      <ul
        className="dropdown-menu"
      >
        <li>
          <a
            onClick=""
          >
            <i
              aria-hidden="true"
              className="fa fa-trash-o dropDownIcon"
            />
            Delete
          </a>
        </li>
        <li>
          <a
            onClick=""
          >
            <i
              aria-hidden="true"
              className="fa fa-pencil-square-o dropDownIcon"
            />
            Rename
          </a>
        </li>
        <li>
          <a
            onClick=""
          >
            <i
              aria-hidden="true"
              className="fa fa-share dropDownIcon"
            />
            Share
          </a>
        </li>
      </ul>
    </div>
    <div
      className="contextmenu dropdown"
      id="contextMenu43970020269"
    >
      <ul
        className="dropdown-menu"
      >
        <li>
           
          <a
            onClick=""
          >
            <i
              aria-hidden="true"
              className="fa fa-trash-o dropDownIcon"
            />
            Delete
          </a>
        </li>
        <li>
          <a
            onClick=""
          >
            <i
              aria-hidden="true"
              className="fa fa-pencil-square-o dropDownIcon"
            />
            Rename
          </a>
        </li>
        <li>
          <a
            onClick=""
          >
            <i
              aria-hidden="true"
              className="fa fa-share dropDownIcon"
            />
            Share
          </a>
        </li>
      </ul>
    </div>
  </td>
  <td
    className="col-xs-1 secure-col"
    style={
      Object {
        "verticalAlign": "middle",
      }
    }
  />
  <td
    className="col-xs-1 "
    style={
      Object {
        "verticalAlign": "middle",
      }
    }
  >
    27.86 KB
  </td>
  <td
    className="col-xs-2 "
    style={
      Object {
        "verticalAlign": "middle",
      }
    }
  >
    1/01/2018 6:46:08 PM
  </td>
</tr>
`;
